### DISCUSSION TITLE
System md   lxc

### DISCUSSION THREAD

**Discussion by bellessecretsshow** [abraunegg/onedrive#2144] [Q&A] [answered]
any help with creating a system md file in lxc  that will run all 6 accounts in monitor mode?  

i tried the  example in the basic set up   --user  but lxc runs only in root       so  all commands are entered apt    rather than sudo apt  

```
root@onedrive:~# systemctl status onedrive@root.service
* onedrive@root.service - OneDrive Free Client for root
     Loaded: loaded (/lib/systemd/system/onedrive@.service; enabled; vendor preset: enabled)
     Active: failed (Result: exit-code) since Wed 2022-09-21 10:39:19 UTC; 2h 56min ago
       Docs: https://github.com/abraunegg/onedrive
    Process: 1082 ExecStart=/usr/bin/onedrive --monitor --confdir=/home/root/.config/onedrive (code=exited, status=3)
   Main PID: 1082 (code=exited, status=3)
        CPU: 52ms

Sep 21 10:39:19 onedrive systemd[1]: Started OneDrive Free Client for root.
Sep 21 10:39:19 onedrive onedrive[1082]: Notification (dbus) server not available, disabling
Sep 21 10:39:19 onedrive onedrive[1082]: Configuring Global Azure AD Endpoints
Sep 21 10:39:19 onedrive onedrive[1082]: Authorize this app visiting:
Sep 21 10:39:19 onedrive onedrive[1082]: https://login.microsoftonline.com/common/oauth2/v2.0/authorize?client_id==Files.Read>
Sep 21 10:39:19 onedrive onedrive[1082]: Enter the response uri: Invalid uri
Sep 21 10:39:19 onedrive onedrive[1082]: Could not initialize the OneDrive API
Sep 21 10:39:19 onedrive systemd[1]: onedrive@root.service: Main process exited, code=exited, status=3/NOTIMPLEMENTED
Sep 21 10:39:19 onedrive systemd[1]: onedrive@root.service: Failed with result 'exit-code'.
```

so i can use onedrive --monitor manually and it starts but i just cant get it to run as a service    ideally i would like just the one container to   monitor 6 different folders to 6 different one drive accounts.     looking at the status output it appears like the systemctl doesnt know its already logged in .

I am really sorry but i am a bit confused   as each account i can start with  
```
onedrive --monitor --confdir="~/.config/onedrive/config"
```

****** UPDATED  **********

Semi Solved   i created a brand new  Systemd    onedrive.service    steps below 
```
$ cd /usr/lib/systemd/system
$ nano onedrive.service
```
This is what i used for the config     --    it might not be ideal for you but it works in proxmox lxc debian 11 
```
[Unit]
Description=OneDrive Free Client for %i
Documentation=https://github.com/abraunegg/onedrive
After=network-online.target
Wants=network-online.target

[Service]
ExecStart=/usr/bin/onedrive --monitor
User=root
Restart=on-failure
RestartSec=3
RestartPreventExitStatus=3

[Install]
WantedBy=multi-user.target
```

save close    ( if you use VIM   i salute you ) 
```
$ systemctl enable onedrive
$ systemctl start onedrive
$ systemctl status onedrive
```
```
root@onedrive:/usr/lib/systemd/system# systemctl status onedrive
* onedrive.service - OneDrive Free Client for 
     Loaded: loaded (/lib/systemd/system/onedrive.service; enabled; vendor preset: enabled)
     Active: active (running) since Wed 2022-09-21 16:12:56 UTC; 6min ago
       Docs: https://github.com/abraunegg/onedrive
   Main PID: 1383 (onedrive)
      Tasks: 9 (limit: 37169)
     Memory: 18.8M
        CPU: 350ms
     CGroup: /system.slice/onedrive.service
             `-1383 /usr/bin/onedrive --monitor

Sep 21 16:12:56 onedrive systemd[1]: Started OneDrive Free Client for .
Sep 21 16:12:56 onedrive onedrive[1383]: Configuration file successfully loaded
Sep 21 16:12:56 onedrive onedrive[1383]: Notification (dbus) server not available, disabling
Sep 21 16:12:56 onedrive onedrive[1383]: Configuring Global Azure AD Endpoints
Sep 21 16:12:58 onedrive onedrive[1383]: Initializing the Synchronization Engine ...
Sep 21 16:12:58 onedrive onedrive[1383]: Initializing monitor ...
Sep 21 16:12:58 onedrive onedrive[1383]: OneDrive monitor interval (seconds): 300
```

i just cant get it to run with 6 different accounts  so i might just create 6 LXCC Debian installs in proxmox as they are only 23mb of ram  with a 16MB/s   upload 

I hope this helps someone   or if someone can explain how i add the other 6 accounts   
??
nano onedrive1.service ??   then config file ??    then systemctl enable onedrive1     i dont know its taken me all day to solve the 1 issue 

??

ok had some sleep and solved adding the other 6  accounts ( amazon do vouchers for family office with 1tb of storage per account  for £65 ) 

after you create the next config  add all the stuff you need   then its just like the above    only give it a different name.     

so  onedrive.service
account2.service
account3.service 

onedrive --confdir="~/.config/account2"

**Comment by abraunegg** [comment, marked as answer]
@bellessecretsshow 
> after you create the next config add all the stuff you need then its just like the above only give it a different name.

Correct .. and this is all documented here: https://github.com/abraunegg/onedrive/blob/master/docs/advanced-usage.md#configuring-the-client-to-use-multiple-onedrive-accounts--configurations
